Call fipstls.Force() when running in FIPS mode (#964)
microsoft/go: eng/doc/fips/README.md

### TLS with FIPS-approved settings

> The Microsoft Go runtime will automatically enforce that `crypto/tls` and `crypto/x509` only use FIPS-approved settings when running in FIPS mode. These differs from upstream's BoringCrypto backend, in which to get these additional restrictions one has to import `crypto/tls/fipsonly`.

Reviewer: I think for now we need to leave the deleted text because this is the live doc for all supported versions.

Once it's backported, I'm wondering now if this kind of change actually might deserve a "since Go 1.19.11, 1.20.6, and 1.21" note in this doc until 1.22 is released.

### TLS with FIPS-approved settings

The Go TLS stack will automatically use OpenSSL crypto primitives. Yet, the FIPS 140-2 standard places additional restrictions on TLS communications, mainly on which cyphers and signers are allowed.

> Since Go 1.22, the Microsoft Go runtime will automatically enforce that `crypto/tls` and `crypto/x509` only use FIPS-approved settings when running in FIPS mode. These differs from upstream's BoringCrypto backend, in which to get these additional restrictions one has to import `crypto/tls/fipsonly`.

Reviewer: A few small things (preferring present tense, using "you" since it's present elsewhere in the doc and "one" hasn't shown up yet, a little rearranging, writing down the reason for the diff):

```suggestion
Since Go 1.22, the Microsoft Go runtime automatically enforces that `crypto/tls` and `crypto/x509` only use FIPS-approved settings when running in FIPS mode. This differs from upstream's BoringCrypto backend, which requires you to import `crypto/tls/fipsonly` to apply the FIPS-mandated restrictions. The Microsoft Go crypto backends do this automatically to reduce the source code changes necessary to produce a FIPS-compliant Go application.
```

> Prior to Go 1.22, a program should import the `crypto/tls/fipsonly` package to configure the Go TLS stack so it is compliant with these restrictions. The configuration is done by an `init()` function, so only importing it is necessary:

Reviewer: I think `must` is nicely direct here if we add the condition (using the TLS stack):

```suggestion
Prior to Go 1.22, a program using the Go TLS stack must import the `crypto/tls/fipsonly` package to be compliant with these restrictions. The configuration is done by an `init()` function, so only importing it is necessary:
```
